Shopify/cli
# Push config draft in dev

### WHY are these changes introduced?

WIP: Missing unit tests

Right now, when you run the `dev` command, `configuration` extensions draft content are not pushed. This means that for instance ui extensions that rely on some configuration value when the `developer preview` is enabled, their preview mode is not working properly



### WHAT is this pull request doing?
- `deploy` rely on the local flag `include_config_on_deploy` to include the config module (and create them if they don't exist) 
- `dev` always includes the config modules as draft extensions (creates the extensions if they don't exist)
- both command should not include the config extensions in case the `versioned_app_beta` is not enabled
- pushed deploy success message is not displayed for `config modules`
- config errors messages are displayed in the `dev console` using a human readable format



### How to test your changes?
- Create an spin instance if you want to verify the database content `spin up partners`. In this case, prepend the CLI commands with `NODE_TLS_REJECT_UNAUTHORIZED=0 SHOPIFY_SERVICE_ENV=spin SPIN_INSTANCE=$(spin show -o name)`
- Create a new remote app `p shopify app config link --path /PATH/TO/YOUR/APP`
- Enable the `versioned_app_config` remote beta
- Run  `p shopify app config link --path /PATH/TO/YOUR/APP`
- Run  `p shopify app deploy --path /PATH/TO/YOUR/APP` and select `no` when you are prompted to include the `configuration`. 
- Optional: Check the database in the table `app_extension_registrations` and you should not see any records for the `client_id` inside the toml
-  Run  `p shopify app dev --path /PATH/TO/YOUR/APP` and wait until the `dev console` is loaded
- Optional: Check the database in the table `app_extension_registrations` and you should see some records for the `client_id` inside the toml. Only with draft_id
- Modify some versioned config value in the `toml`. You should not see any message in the `dev console`
- Optional: Check the database in the table `app_static_extension_config` the records linked to the registrations id inside the table `app_extension_registrations`. Check the value of the config column to verify it contains the correct value
-  Modify some versioned config value in the `toml` using a wrong value. You should an error message like this
![image](https://github.com/Shopify/cli/assets/105213827/5f30fcc0-a4e4-4301-b6de-bb9b4ce064e6)

- Optional: Check the database in the table `app_static_extension_config` the records linked to the registrations id inside the table `app_extension_registrations`. Check the value hasn't been modified
- Restore the correct  last value in the `toml`
- Run `p shopify app config link --path /PATH/TO/YOUR/APP`. The modified value will be restored with the default value as the modification has only been made in the draft content
- Stop the `dev` process and run `p shopify app deploy --path /PATH/TO/YOUR/APP --reset`. Select the same remote app and this time confirm the prompt to include the configuration. 
- Modify something in the `toml`
- Run `p shopify app config link --path /PATH/TO/YOUR/APP`. The modified value will be restored with the last deployed configuration

### Post-release steps



### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
